src/main/java/bio/overture/songsearch/config/websecurity/AuthDisabledConfig.java

```diff
@@ -0,0 +1,23 @@
+package bio.overture.songsearch.config.websecurity;
+
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Profile;
+import org.springframework.security.config.annotation.web.reactive.EnableWebFluxSecurity;
+import org.springframework.security.config.web.server.ServerHttpSecurity;
+import org.springframework.security.web.server.SecurityWebFilterChain;
+
+
+@EnableWebFluxSecurity
+@Profile("!secure")
+public class AuthDisabledConfig {
+    @Bean
+    public SecurityWebFilterChain securityFilterChain(
+            ServerHttpSecurity http) {
+        http
+                .csrf().disable()
+                .authorizeExchange()
+                .pathMatchers("/graphql/**").permitAll()
+                .pathMatchers("/actuator/**").permitAll();
+        return http.build();
+    }
+}
```

Comment:
this could be merged into a single `AuthConfig` class using profiles on the correct methods

Reply:
following additional discussion, this comment is withdrawn